src/script/api/script_airport.cpp

```diff
@@ -136,7 +137,7 @@
 	const AirportSpec *as = ::AirportSpec::Get(type);
 	if (!as->IsWithinMapBounds(0, tile)) return -1;
 
-	if (_settings_game.economy.station_noise_level) {
+	if (_settings_game.economy.station_noise_level || type == AT_OILRIG) {
```

Comment:
What's special about oil rigs that they always need to go into this branch?

Reply:
They don't produce noise.

Reply:
https://gist.github.com/SamuXarick/db932246d86aecd117affca3c023d93e

![image](https://github.com/OpenTTD/OpenTTD/assets/43006711/f047cf86-1ade-4496-b5c4-67a7f871643b)